subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1328,6 +1486,45 @@ SHELL_STATIC_SUBCMD_SET_CREATE(wifi_commands,
 		"-f: Force to use this regulatory hint over any other regulatory hints\n"
 		"Note: This may cause regulatory compliance issues, use it at your own risk.",
 		cmd_wifi_reg_domain),
+	SHELL_CMD(mode, NULL, "mode operational setting\n"
+		"parameters:\n"
+		"<Interface index: interface index of the interface>\n"
+		"<Mode set value: bitmap of values.>\n"
+		"Multiple values can be set simultaneously\n"
+		"the mode support check is to be performed by lower layers\n"
+		"bit(0) = Station mode\n"
+		"bit(1) = Monitor mode\n"
+		"bit(2) = TX-Injection mode\n"
+		"bit(3) = Promiscuous mode\n"
+		"bit(4) = AP mode\n"
+		"bit(5) = STA-AP mode\n"
+		"Usage: Get operation - wifi mode <if_idx>\n"
+		"Set operation - wifi mode <if_idx> <bitmap value in hexadecimal>\n",
+		cmd_wifi_mode),
+	SHELL_CMD(packet_filter, NULL, "mode filter setting\n"
+		"This command is used to set packet filter setting when\n"
+		"monitor, TX-Injection and promiscuous mode is enabled.\n"
+		"parameters:\n"
+		"<Interface index: interface index of the interface.>\n"
+		"<Filter set value: bitmap of values.>\n"
+		"bit(0) = ALL packet types are allowed up the stack\n"
+		"bit(1) = Management packets are allowed up the stack\n"
+		"bit(2) = Data packets are allowed up the stack\n"
+		"bit(3) = control packets are allowed up the stack\n"
```

Comment:
Same question for the packet types, are all the bit combinations allowed at the same time?

Reply:
Ye. It is possible to set a combination of packet filter types. The reason that bit(0) is provided as "all the different filters are enabled" is to provide a way to define all filters by one bit mapping. As, lower layers will not have to check the same and directly enable all packet filter settings when bit(0) is only set.